src/pedon/soil.py

```diff
@@ -396,6 +492,319 @@ def rosetta(self, version: Literal[1, 2, 3] = 3) -> Genuchten:
             n=10 ** vgpar[3],
         )
 
+    def hypags(self) -> Genuchten:
+        """
+        Implement the HYPAGS procedures to estimate van Genuchten parameters and
+        characteristic grain-size metrics from hydraulic conductivity and/or grain
+        size inputs. The routine implements the Kozeny-Carman-based
+        parameterization from Peche & Houben (2023, 2024) and:
+        - Accepts as inputs: k (preferred) or d10 or d20 (grain diameters in m).
+        - Iteratively estimates effective porosity (ne), d50, and d60 from the
+        Kozeny-Carman relationships.
+        - Estimates a capillary-rise representative grain diameter and computes
+        alpha and n (van Genuchten parameters) with bootstrap-derived error
+        ranges available in the original algorithm.
+        - Genuchten-like object with k_s, alpha, n populated. theta_r and theta_s
+        may be left None because HYPAGS focuses on k/alpha/n estimation.
+        Raises
+        - The method expects physical units consistent with the HYPAGS formulation:
+        grain diameters in meters, k in m s^-1 (or converted internally as
+        described in the code). Input values outside empirically supported ranges
+        will trigger warnings but the routine will still attempt computation.
+
+        References
+        ----------
+            - Peche, A., Houben, G., & Altfelder, S. (2024). Approximation of van Genuchten Parameter Ranges from Hydraulic Conductivity Data. Groundwater, 62(3), 469-479.
+            - Peche, A., & Houben, G. J. (2023). Estimating characteristic grain sizes and effective porosity from hydraulic conductivity data. Groundwater, 61(4), 574-585.
+        """
+        # constants and coefficients
+        rho_f = 999.7  # fluid density [kg/m³] (assumed 20°C)
+        g = 9.81  # gravitational acceleration [m/s²]
+        mu = 1.1306e-3  # dynamic viscosity [kg/(m·s)]
+        c = mu / (rho_f * g)
+        a1, a2, a3 = 1.00401066e00, 1.50991411e-04, 5.78888587e-03  # alpha-coefficients
+        gamma = 0.0728  # surface tension [J/m²]
+        theta = 0.0  # contact angle, parameter for Young-Laplace eq.
+        factor = (
+            2 * gamma * cos(theta) / (rho_f * g)
+        )  # factor Young-Laplace eq. (fluid properties, contact angle, surface tension)
+        crgd1, crgd2 = (
```

Comment:
these constants aren't used. can they be removed? or are they needed for something else?